File: eecs268/2021fall/MWF12pm/2021.09.03/Dog.cpp
#include "Dog.h"

void Dog::doTrick()
{
	std::cout << "Trick occuring.\n";
}

void Dog::eat()
{
	doTrick();
	std::cout << "Dog eating...\n";
}

void Dog::sleep()
{
	std::cout << "Dog sleeping...\n";
}

File: eecs268/2025fall/TR/2025.12.02.inheritance/inheritance/dog.py
#dog.py

from animal import Animal

class Dog(Animal):
    def __init__(self):
        super().__init__()
        print('Dog created')
        self.breed = 'UNDEFINED'
    
    def do_trick(self):
        print('Dog doing an amazing trick')

    def sleep(self):
        print('Dog sleeping. Hrp hrp hrp HRP')
